core/eolearn/core/eodata.py

```diff
@@ -461,7 +461,7 @@ def __copy__(self, features: FeaturesSpecification = ...) -> EOPatch:
         if not features:  # For some reason deepcopy and copy pass {} by default
             features = ...
 
-        new_eopatch = EOPatch()
+        new_eopatch = EOPatch(bbox=copy.copy(self.bbox))
```

Comment:
potential idea if we are going to be doing this a lot, it would perhaps make sense that we add a `.copy(deep: bool = False)` method to the `BBox` class, perhaps also the `Geometry`.

If this is more of a one and done thing, it's fine as is.

Reply:
i what custom behaviour would you want there that is not provided by `copy.copy`?
I think a better question is if `EOPatch` actually needs custom implementations of `__copy__` and `__deepcopy__` or if the default behaviour is good enough. The `FeatureIO` objects complicate stuff a bit i believe.

Reply:
You can ignore my initial comment, but can you explain how do the custom implementations of `__copy__` and `__deepcopy__` differ from the default ones?

Reply:
The main difference is that we ensure that `FeatureIO` objects are not loaded, but also not deep-copied (because they have a filesystem attribute :/ copying uses pickling behind the scenes), so we do a workaround for that